### PR TITLE
WIP: Map::setNode: Remove unnecessary exceptions

### PR DESCRIPTION
**Pull request by osjc**
Clean up unnecessary exception usage and handling from Map::setNode and all code using it.

### CODE REVIEW

**Comment by osjc** [comment]
Fixed some questionable code formatting and a factually incorrect comment.

File: src/map.cpp [hunk 1]
```diff
@@ -122,43 +122,35 @@ MapSector * Map::getSectorNoGenerate(v2s16 p)
 	return getSectorNoGenerateNoLock(p);
 }
 
-MapBlock * Map::getBlockNoCreateNoEx(v3s16 p3d)
+MapBlock * Map::getBlockNoCreate(v3s16 p3d)
```
Comment (ClobberXD) [inline src/map.cpp]:
```suggestion
MapBlock *Map::getBlockNoCreate(v3s16 p3d)
```

File: src/map.cpp [hunk 2]
```diff
@@ -122,43 +122,35 @@ MapSector * Map::getSectorNoGenerate(v2s16 p)
...
 {
 	v2s16 p2d(p3d.X, p3d.Z);
 	MapSector * sector = getSectorNoGenerate(p2d);
```
Comment (ClobberXD) [inline src/map.cpp]:
```suggestion
	MapSector *sector = getSectorNoGenerate(p2d);
```

PR doesn't touch this line, but it'd be great if this minor nit could be fixed.

File: src/map.cpp [hunk 3]
```diff
@@ -122,43 +122,35 @@ MapSector * Map::getSectorNoGenerate(v2s16 p)
...
 	if(sector == NULL)
 		return NULL;
```
Comment (ClobberXD) [inline src/map.cpp]:
```cpp
if (!sector)
	return nullptr;
```

File: src/map.cpp [hunk 4]
```diff
@@ -122,43 +122,35 @@ MapSector * Map::getSectorNoGenerate(v2s16 p)
...
+	MapBlock *block = getBlockNoCreate(blockpos);
 	return block && block->getIsUnderground(); 
 }
 
 bool Map::isValidPosition(v3s16 p)
 {
 	v3s16 blockpos = getNodeBlockPos(p);
-	MapBlock *block = getBlockNoCreateNoEx(blockpos);
+	MapBlock *block = getBlockNoCreate(blockpos);
 	return (block != NULL);
```
Comment (ClobberXD) [inline src/map.cpp]: This method can be condensed to one line:

```cpp
bool Map::isValidPosition(v3s16 p)
{
	return getBlockNoCreate(getNodeBlockPos(p));
}
```

If I'm not mistaken, this method should return `true` if the returned expression is anything other than `nullptr`, and `false` if `nullptr`.

File: src/map.cpp [hunk 5]
```diff
@@ -2072,7 +2061,7 @@ void ServerMap::loadBlock(std::string *blob, v3s16 p3d, MapSector *sector, bool
 
 MapBlock* ServerMap::loadBlock(v3s16 blockpos)
 {
-	bool created_new = (getBlockNoCreateNoEx(blockpos) == NULL);
+	bool created_new = (getBlockNoCreate(blockpos) == NULL);
```
Comment (ClobberXD) [inline src/map.cpp]:
```suggestion
	bool created_new = !getBlockNoCreate(blockpos);
```

File: src/map.h [hunk 6]
```diff
@@ -167,14 +167,14 @@ class Map /*: public NodeContainer*/
...
 	virtual MapSector * emergeSector(v2s16 p){ return NULL; }
 
-	// Returns InvalidPositionException if not found
-	MapBlock * getBlockNoCreate(v3s16 p);
 	// Returns NULL if not found
-	MapBlock * getBlockNoCreateNoEx(v3s16 p);
+	MapBlock * getBlockNoCreate(v3s16 p);
 
 	/* Server overrides */
 	virtual MapBlock * emergeBlock(v3s16 p, bool create_blank=true)
```
Comment (ClobberXD) [inline src/map.h]:
```suggestion
	virtual MapBlock *emergeBlock(v3s16 p, bool create_blank = true)
```

Again, untouched code, but would be nice if fixed :)

File: src/mapsector.cpp [hunk 7]
```diff
@@ -67,7 +67,7 @@ MapBlock * MapSector::getBlockBuffered(s16 y)
 	return block;
 }
 
-MapBlock * MapSector::getBlockNoCreateNoEx(s16 y)
+MapBlock * MapSector::getBlockNoCreate(s16 y)
```
Comment (ClobberXD) [inline src/mapsector.cpp]:
```suggestion
MapBlock *MapSector::getBlockNoCreate(s16 y)
```

File: src/mapsector.h [hunk 8]
```diff
@@ -50,7 +50,7 @@ class MapSector
 		return m_pos;
 	}
 
-	MapBlock * getBlockNoCreateNoEx(s16 y);
+	MapBlock * getBlockNoCreate(s16 y);
```
Comment (ClobberXD) [inline src/mapsector.h]:
```suggestion
	MapBlock *getBlockNoCreate(s16 y);
```

Please also fix this in the following lines, if possible.

**Comment by osjc** [comment]
Changed to WIP because of multiple "X * Y" lines that are touched with this. Going to create a different PR that will get rid of all of these, then rebase this PR on top of that, then remove WIP.

**Comment by osjc** [comment]
Blocked by the need to do a more thorough coding standards fixup, which will take a lot of time to come. So closing for now. Will reopen once I am ready to finish this.

**Pull request closed without merging**